lib/V8/v8-utils.cpp

```diff
@@ -4416,14 +4466,23 @@ static void JS_ExecuteExternal(v8::FunctionCallbackInfo<v8::Value> const& args)
     }
   }
 
-  bool usePipes = false;
+  auto workingDirectory = FileUtils::currentDirectory().result();
+  std::string subProcessWorkingDirectory = workingDirectory;
 
-  if (3 <= args.Length()) {
-    usePipes = TRI_ObjectToBoolean(isolate, args[2]);
-  }
+  if (5 <= args.Length()) {
+    TRI_Utf8ValueNFC name(isolate, args[4]);
+    if (*name == nullptr) {
+      TRI_V8_THROW_TYPE_ERROR("<workingDirectory> must be a string");
+    }
 
+    subProcessWorkingDirectory = std::string(*name, name.length());
+  }
   ExternalId external;
+  if (subProcessWorkingDirectory != workingDirectory) {
+    FileUtils::changeDirectory(subProcessWorkingDirectory);
+  }
   TRI_CreateExternalProcess(*name, arguments, additionalEnv, usePipes, &external);
+  FileUtils::changeDirectory(workingDirectory);
```

Comment:
This is still unsafe, because cd-ing somewhere else for a short while in a multi-threaded process...
But anyway, the function is deactivated by default so at least one can only do harm when setting certain start parameters.